packages/lesswrong/lib/collections/moderatorActions/helpers.ts

```diff
@@ -1,5 +1,42 @@
 import moment from "moment";
 import { forumTypeSetting } from "../../instanceSettings";
+import ModeratorActions from "./collection";
+import { rateLimits, RateLimitType, RATE_LIMIT_ONE_PER_DAY, RATE_LIMIT_ONE_PER_FORTNIGHT, RATE_LIMIT_ONE_PER_MONTH, RATE_LIMIT_ONE_PER_THREE_DAYS, RATE_LIMIT_ONE_PER_WEEK } from "./schema";
+
+/**
+ * For a given RateLimitType, returns the number of hours a user has to wait before posting again.
+ */
+export function getTimeframeForRateLimit(type: RateLimitType) {
+  let hours 
+  switch(type) {
+    case RATE_LIMIT_ONE_PER_DAY:
+      hours = 24; break;
+    case RATE_LIMIT_ONE_PER_THREE_DAYS:
+      hours = 24 * 3; break;
+    case RATE_LIMIT_ONE_PER_WEEK:
+      hours = 24 * 7; break;
+    case RATE_LIMIT_ONE_PER_FORTNIGHT:
+      hours = 24 * 14; break;
+    case RATE_LIMIT_ONE_PER_MONTH:
+      hours = 24 * 30
+  }
+  return hours
+}
+
+/**
+ * Fetches the most recent, active rate limit affecting a user.
+ */
+export async function getModeratorRateLimit(user: DbUser) {
+  return await ModeratorActions.findOne({
+    userId: user._id,
+    type: {$in: rateLimits},
```

Comment:
One slightly annoying thing is that specifying the `type` selector like this doesn't actually narrow down the return type signature to preserve that information at all, by itself.  I think it probably makes sense to preserve that info in the return type signature of this function, to avoid forcing callers to perform manual type casts, as you do in the [modified post and comment rate limit logic](https://github.com/ForumMagnum/ForumMagnum/pull/6057/files#diff-712b78e6fe18e483bef81886d917ba682cba2c6b08f39cdd86d1433cf7e2640dR57).  One way would just be to perform the type cast here.  Another would be to actually run the type guard against the result before returning it, though that's sort of a redundant operation (that might prevent us from getting back some invalid moderator action result of there's, like, a database bug which causes that part of the selector to not work correctly, or something, but doesn't protect against much else).